sqlmesh/core/snapshot/definition.py

```diff
@@ -1051,8 +1047,18 @@ def audit(self) -> StandaloneAudit:
 
     @property
     def depends_on_past(self) -> bool:
-        """Whether or not this models depends on past intervals to be accurate before loading following intervals."""
-        return self.is_model and self.model.depends_on_past
+        """Whether or not this models depends on past intervals to be populated before loading following intervals.
```

Comment:
can you add the specific example about incremental by time with a self reference not fitting in full_history_restatement, that is why we need to do the OR here

Reply:
Added an example.